### PR TITLE
Add unit tests to identify configuration migration scenarios for checkpointing

### PR DESCRIPTION
**Pull request by baijumeswani**
**Description**: This pull request adds unit test coverage for checkpointing.
These tests help identify working configuration migration checkpoint scenarios, and will help ensure that the checkpoint feature is not impacted by any fresh changes.

Unit tests can be broken down into following categories:
- Checkpoint going from all training configurations to a single node training configuration.
- Checkpoint going from all training configurations to a data parallel training configuration.
- Checkpoint going from all training configurations to a distributed zero training configuration.

In summary, this pull request tries to identify which trainer migrations are currently supported by checkpointing.

### CODE REVIEW

File: tools/ci_build/github/linux/docker/scripts/install_deps.sh [hunk 1]
```diff
@@ -113,9 +113,7 @@ if [ $DEVICE_TYPE = "gpu" ]; then
     if [[ $BUILD_EXTR_PAR = *--enable_training* ]]; then
       ${PYTHON_EXE} -m pip install -r ${0/%install_deps.sh/training\/requirements.txt}
 
-      if [[ $BUILD_EXTR_PAR = *--enable_training_python_frontend_e2e_tests* || $BUILD_EXTR_PAR = *enable_training_pipeline_e2e_tests* ]]; then
-        source ${0/%install_deps.sh/install_openmpi.sh}
-      fi
+      source ${0/%install_deps.sh/install_openmpi.sh}
```
Comment (thiagocrepaldi) [inline tools/ci_build/github/linux/docker/scripts/install_deps.sh:116]: Why are we removing that constraint?

**Comment by thiagocrepaldi** [comment]
We have to add this to the nightly pipeline too

File: tools/ci_build/github/linux/docker/scripts/install_deps.sh (hunk 1 above)
Comment (baijumeswani) [reply tools/ci_build/github/linux/docker/scripts/install_deps.sh:116]: I need openmp for functions such as ```get_mpi_context_world_rank()``` and ```get_mpi_context_world_size()```. Otherwise, this errors out for me.

Comment (thiagocrepaldi) [reply tools/ci_build/github/linux/docker/scripts/install_deps.sh:116]: Check with Liqun if we are allowe dto do that. Otherwise, we will install MPI in VMs that we shouldnt.

If you are getting this for the nightly pipeline, than something is wrong. If you are using a non-nightly, VM, why do we need MPI? MPI is for distributed training and "regular" VMs do not support it, so it doesnt make sense to install MPI on them

Comment (baijumeswani) [reply tools/ci_build/github/linux/docker/scripts/install_deps.sh:116]: you're right. It is not needed for non distributed scenario. Will remove it from the script since we are not going to run this test for the on commit verify job.

**Comment by baijumeswani** [comment]
Added the test to nightly pipeline

**Review by thiagocrepaldi — CHANGES_REQUESTED** [review]
The YAML file has to be checked in separately, then added to the github CI by an admin and only then PR with tests can be checked in. Otherwise, the tests go in without running/being verified

**Comment by baijumeswani** [comment]
The azure distributed pipeline can only run for new pull requests. So this one won't run automatically.
I manually triggered the [azure pipeline here](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=265496&view=logs&j=8e52c8ec-d466-52da-f78a-ae1422442b01).

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py [hunk 2]
```diff
@@ -0,0 +1,1475 @@
...
+import torch
+import torch.distributed as dist
+
+import onnxruntime
+from onnxruntime import set_seed
+from onnxruntime.training import amp, checkpoint, optim, orttrainer
+from onnxruntime.capi._pybind_state import set_cuda_device_id, get_mpi_context_world_rank, get_mpi_context_world_size
+from orttraining_test_orttrainer_frontend import _load_pytorch_transformer_model
+
+def distributed_setup(save_function):
```
[the remark this answers is not in the document]
Comment (thiagocrepaldi) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py:21]: If this is a helper method, create a file such as `checkpoint/_tests_helpers.py` and move everything that is not needed for the checkpoint there. This way, users can quick identify what is the bae minimum to use your sample without adding dead code as the env environemnts below

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py [hunk 3]
```diff
@@ -0,0 +1,1475 @@
...
+    # compare all states
+    # model states
+    for key, value in state_dict_post_checkpoint['fp16_param'].items():
+        assert_allclose(value, state_dict_pre_checkpoint['fp32_param'][key[:-5]], atol=1e-3)
+
+    # compare optimizer states
+    for key, value in state_dict_pre_checkpoint['optimizer'].items():
+        assert_allclose(value, state_dict_post_checkpoint['optimizer'][key])
+
+if __name__ == '__main__':
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py:1428]: Do we need more than one entry point for testing?

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py [hunk 4]
```diff
@@ -0,0 +1,1475 @@
...
+
+    # run an eval step to innitialize the graph
+    torch.manual_seed(seed)
+    set_seed(seed)
+    data, targets = batcher_fn(train_data, 0)
+    trainer.eval_step(data, targets)
+
+    return checkpoint.experimental_state_dict(trainer), model
+
+def split_state_dict(state_dict):
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py:60]: There are some scenarios that cannot be tested directly; for example going from a single node run to a distributed run. Workflow that illustrates why this scenario cannot be tested directly.
- Two files are saved through ```orttraining_test_save_checkpoint.py```
  - state dictionary through pickle after being extracted through ```experimental_state_dict``` method
  - checkpoint file through ```experimental_save_ceckpoint```
- Loading is done through:
  - ```experimental_load_checkpoint``` for ORTTrainer
  - pickle load for pre checkpoint state dictionary
- now the idea is to compare the state dictionary extracted through ```experimental_state_dict``` on the checkpoint loaded ORTTrainer and the state dictionary loaded through pickle.
- But these two files cannot be directly compared because one is a single node run where as the other one is a distributed run.
- So, I need to aggregate the state dictionaries for the distributed run and then compare against the originally loaded state dictionary.

These functions are the aggregation functions that are similar to the ones defined in the new aggregation logic in #5478

That is why these functions are needed. I can add docstrings to these functions, but these are temporary functions that I think will need to be removed once #5478 is merged.

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py [hunk 5]
```diff
@@ -0,0 +1,1475 @@
...
+    optim_config = optim.LambConfig(lr=learning_rate)
+    model, model_desc, loss_fn, batcher_fn, train_data, _, _ = _load_pytorch_transformer_model(device)
+    trainer = orttrainer.ORTTrainer(model, model_desc, optim_config, loss_fn=loss_fn, options=trainer_opts)
+
+    # load checkpoint into trainer
+    checkpoint.experimental_load_checkpoint(trainer, checkpoint_dir)
+
+    # run an eval step to innitialize the graph
+    torch.manual_seed(seed)
+    set_seed(seed)
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py:54]: I have ```set_seed``` before instantiating the ```ORTTrainer``` as well. Through testing I realized that I need to set seed before instantiating the ```ORTTrainer``` as well as before calling ```eval_step``` function.

File: orttraining/orttraining/test/python/orttraining_test_checkpoint.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    if not os.path.exists(checkpoint_dir):
+        os.makedirs(checkpoint_dir, exist_ok = True)
+
+if __name__ == '__main__':
+    checkpoint_dir = 'checkpoint_dir/'
+    if not os.path.exists(checkpoint_dir):
+        os.makedirs(checkpoint_dir, exist_ok = True)
+
+    ngpus = torch.cuda.device_count()
+
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_checkpoint.py:21]: Will add a comment with the workflow.

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py [hunk 7]
```diff
@@ -0,0 +1,1475 @@
...
+        device = 'cuda:' + str(world_rank)
+
+        os.environ['RANK'] = str(world_rank)
+        os.environ['WORLD_SIZE'] = str(world_size)
+        os.environ['MASTER_ADDR'] = '127.0.0.1'
+        os.environ['MASTER_PORT'] = '29500'
+
+        set_cuda_device_id(world_rank)
+
+        dist.init_process_group(backend='nccl', world_size=world_size, rank=world_rank)
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py:34]: Lines 27-34 are needed only for test purposes. I will leave a comment so it is clear.

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py:21]: will do that.

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py (hunk 3 above)
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py:1428]: I don't think we would have more than 1 entry point. I can remove the ```__main__``` check.

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (thiagocrepaldi) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py:60]: That is great information, please add this explanation on why it is needed, how you solved and any limitation that we still have after all workarounds. That helps determining how covered we are and how to debug when something unexpected happens

[the remark this answers is not in the document]
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py]: Added detailed workflow comment in the main test entry point

File: orttraining/orttraining/test/python/orttraining_test_checkpoint.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_checkpoint.py:24]: Added a detailed comment.

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py]: Moved this test infra setup into a function of it's own so it is clear that it is only for testing purposes.

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py (hunk 3 above)
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py]: Removed ```__main__``` check for all files other than the main entry point (```orttraining_test_checkpoint.py```). I was contemplating whether or not to remove from there as well, and decided to let that one be there.

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py]: I added a docstring and moved helper functions to ```checkpoint/_test_helpers.py```.

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py [hunk 8]
```diff
@@ -0,0 +1,1384 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:4]:
```suggestion
################################################################################
# Refer to orttraining_test_checkpoint.py for an overview about Checkpoint tests
################################################################################

```

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_save_checkpoint.py [hunk 9]
```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/checkpoint/orttraining_test_save_checkpoint.py:5]:
```suggestion
################################################################################
# Refer to orttraining_test_checkpoint.py for an overview about Checkpoint tests
################################################################################

import os
```

File: orttraining/orttraining/test/python/orttraining_test_checkpoint.py [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+import shutil
+import sys
+import torch
+from checkpoint._test_helpers import makedir
+
+def _single_run(execution_file, scenario, checkopint_dir):
+    assert subprocess.call([sys.executable, execution_file, '--scenario', scenario, '--checkpoint_dir', checkopint_dir]) == 0
+
+def _distributed_run(execution_file, scenario, checkopint_dir):
+    assert subprocess.call(['mpirun', '-n', str(ngpus), '-x', 'NCCL_DEBUG=INFO', sys.executable, execution_file, '--scenario', scenario,  '--checkpoint_dir', checkopint_dir]) == 0
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_checkpoint.py:16]: Excellent!!! :)

File: orttraining/orttraining/test/python/orttraining_test_checkpoint.py [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+    #             - equality (or new equality) of optimizer states
+    #         - In some cases the comparison is not directly possible; for example single node trainer to a distributed zero trainer because the extracted state
+    #           dictionary is a distributed one and cannot be compared against a single node trainer directly.
+    #             - First these states are saved using pickle for each rank to a file on disk
+    #             - Wait for all ranks to complete writing the file to disk using barrier()
+    #             - Load all states and aggregate them into 1 state dictionary
+    #             - Compare this aggregated state dictionary against the original one loaded from disk.
+
+    save_checkpoint_file = 'checkpoint/orttraining_test_save_checkpoint.py'
+    load_checkpoint_file = 'checkpoint/orttraining_test_load_checkpoint.py'
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_checkpoint.py:59]:
```suggestion
    save_checkpoint_file = os.path.join('checkpoint','orttraining_test_save_checkpoint.py')
    load_checkpoint_file = os.path.join('checkpoint','orttraining_test_load_checkpoint.py')
```

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py [hunk 12]
```diff
@@ -0,0 +1,1384 @@
...
+from _test_helpers import distributed_setup, load_evaluate_extract_state_dict, split_state_dict, split_name, aggregate_states
+
+def test_load_from_single_node_full_precision_into_single_node_full_precision(device = 'cuda', checkpoint_dir = 'checkpoint_dir/single_node/full_precision/'):
+    opts = {'device' : {'id' : device},
+            'debug' : {'deterministic_compute': True}}
+    
+    # extract state dictionaries to compare
+    state_dict_post_checkpoint, model = load_evaluate_extract_state_dict(device, orttrainer.ORTTrainerOptions(opts), checkpoint_dir)
+
+    state = pickle.load(open(os.path.join(checkpoint_dir, 'state_dict.pkl'), 'rb'))
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:26]: Dont you have to call `close()` on the file handle to prevent memory leak?
```suggestion
    with open(os.path.join(checkpoint_dir, 'state_dict.pkl'), 'rb') as f:
        state = pickle.load(f)
```

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py [hunk 13]
```diff
@@ -0,0 +1,1384 @@
...
+    state_dict_pre_checkpoint = state['state_dict']
+
+    # compare all states
+    for key, value in state_dict_pre_checkpoint.items():
+        if key.endswith('_fp16'):
+            full_precision_key = key[:-5]
+            assert full_precision_key in state_dict_post_checkpoint
+            assert_allclose(value, state_dict_post_checkpoint[full_precision_key], atol=1e-3)
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:54]: We should probably define a global tolerance for fp16 and another for fp32 and use the same across all tests
Previous experience showed that people will fix specific tests and CI will fail many times before all tests are fixed, one by one

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py [hunk 14]
```diff
@@ -0,0 +1,1384 @@
...
+
+    # compare all states
+    for key, value in state_dict_pre_checkpoint.items():
+        assert_allclose(value, state_dict_post_checkpoint[key])
+    
+    # load state into pytorch and compare
+    model.load_state_dict(state_dict_pre_checkpoint, strict=False)
+    state_dict_pytorch = model.state_dict()
+    for key, value in state_dict_pytorch.items():
+        assert_allclose(value, state_dict_pre_checkpoint[key])
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:37]: optional: This snippet seems to be repeated in most tests. maybe helper function for each big scenario might help

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py [hunk 15]
```diff
@@ -0,0 +1,1384 @@
...
+
+def test_load_from_single_node_mixed_precision_into_single_node_mixed_precision(device = 'cuda', checkpoint_dir = 'checkpoint_dir/single_node/mixed_precision/'):
+    opts = {
+                'device' : {'id' : device},
+                'mixed_precision':
+                {
+                    'enabled': True,
+                    'loss_scaler': amp.DynamicLossScaler()
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:70]:
```suggestion
```

when ` enabled=True` and `loss_scaler`  is not specified, `amp.DynamicLossScaler()`  is assumed as default

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py [hunk 16]
```diff
@@ -0,0 +1,1384 @@
...
+            assert_allclose(value, state_dict_pre_checkpoint[full_precision_key], atol=1e-3)
+        else:
+            assert_allclose(value, state_dict_pre_checkpoint[key])
+
+def test_load_from_data_parallelism_full_precision_into_single_node_full_precision(device = 'cuda', checkpoint_dir = 'checkpoint_dir/data_parallelism/full_precision/'):
+    opts = {'device' : {'id' : device},
+            'debug' : {'deterministic_compute': True}}
+    
+    # extract state dictionaries to compare
+    state_dict_post_checkpoint, model = load_evaluate_extract_state_dict(device, orttrainer.ORTTrainerOptions(opts), checkpoint_dir)
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:116]: I noticed that ` orttrainer.ORTTrainerOptions(opts)` is called for every test. Consider passing just  `opts`  and the method itself instantiates ` ORTTrainerOption`

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py [hunk 17]
```diff
@@ -0,0 +1,1384 @@
...
+
+            # collect optimizer states for later comparison since they are sharded
+            aggregate_states(optimizer_states, rank_state_dict['optimizer'])
+
+        # compare optimizer states
+        """
+        failure_case: non sharded to sharded weights fails
+        for key, value in optimizer_states.items():
+            assert_allclose(value.reshape(state_dict_pre_checkpoint['optimizer'][key].size()), state_dict_pre_checkpoint['optimizer'][key])
+        """
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:802]: What does this mean? This test would fail because of a bug?

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_save_checkpoint.py [hunk 18]
```diff
@@ -0,0 +1,222 @@
...
+    # PyTorch transformer model as example
+    opts = {'device' : {'id' : device},
+            'debug' : {'deterministic_compute': True}}
+    opts = orttrainer.ORTTrainerOptions(opts)
+    optim_config = optim.LambConfig(lr=learning_rate)
+    model, model_desc, loss_fn, batcher_fn, train_data, _, _ = _load_pytorch_transformer_model(device)
+    trainer = orttrainer.ORTTrainer(model, model_desc, optim_config, loss_fn=loss_fn, options=opts)
+
+    # run train steps
+    train(trainer, train_data, batcher_fn)
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/checkpoint/orttraining_test_save_checkpoint.py:33]: Just like you did for the load part, put all this in a helper and save lots of typing :)

**Review by thiagocrepaldi — CHANGES_REQUESTED** [review]

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py (hunk 12 above)
Comment (baijumeswani) [reply orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:26]: yeah, I need to use with open. I forgot to change that. Updating.

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py (hunk 17 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:805]: Ideally we dont let codes commented out as they get stale and forgotten. But as this will be short-lived, you can convert it into a TODO (like  TODO: Uncomment this after Checkpoint redesign. Current implementation does not support it) and let it commented out

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py (hunk 14 above)
Comment (baijumeswani) [reply orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:45]: each test may have slightly different testing condition. Leaving this as is.

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py (hunk 17 above)
Comment (baijumeswani) [reply orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py]: Added a TODO there.

File: orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py (hunk 3 above)
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_load_checkpoint.py]: Updated ```orttraining_test_checkpoint.py``` file as well.

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py [hunk 19]
```diff
@@ -0,0 +1,1472 @@
...
+import torch.distributed as dist
+
+import sys
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+import onnxruntime
+from onnxruntime.training import checkpoint
+from _test_helpers import distributed_setup, create_orttrainer_and_load_checkpoint, split_state_dict, aggregate_states
+
+global_fp16_fp32_atol = 1e-3
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:26]: You are only using this tolerance for fp16. When you omit `atol` from `assert_allclose` for fp32 tests, for example, `1e-7` is assumed

Let's merge like this, but in a future PR, let's either create a FP32_ATOL and FP16_ATOL or just rename this one to global_fp16_atol and use the 1e-7 for fp32

**Review by thiagocrepaldi — APPROVED** [review]

File: orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py (hunk 19 above)
Comment (baijumeswani) [reply orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:26]: So, ```global_fp16_fp32_atol``` atol is not for fp16 or fp32, but for comparison between fp16 and fp32. Any comparison between two fp16s is done using ```atol=1e-7``` which is the default value. Similarly, for all comparisons between two fp32s is again done using ```atol=1e-7```.

Comment (baijumeswani) [reply orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:26]: Having said that, maybe it could have been explicitly set. Will update in the next pull request.

Comment (thiagocrepaldi) [reply orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py:26]: I think it is a good name then. Let it be

**Comment by baijumeswani** [comment]
Thank you for the review 😄
